### PR TITLE
Collect metrics from Director

### PR DESCRIPTION
**Pull request by pkosiec**
**Description**

Changes proposed in this pull request:

- Collect metrics from Director

**Related issue(s)**
#1235

### CODE REVIEW

File: components/director/cmd/director/main.go [hunk 1]
```diff
@@ -281,3 +303,22 @@ func getRuntimeMappingHanderFunc(transact persistence.Transactioner, cachePeriod
...
 }
+
+func createMetricsServer(metricsAddress string) *http.Server {
+	metricsHandler := http.NewServeMux()
+	metricsHandler.Handle("/metrics", promhttp.Handler())
+
+	return &http.Server{Addr: metricsAddress, Handler: metricsHandler}
+}
+
+func runServer(srv *http.Server, name string) {
```
Comment (crabtree) [inline components/director/cmd/director/main.go:314]: Maybe this function can return the shutdown function for the given server and name? Right now you need to call `runServer` and `shutdownServer` with similar parameters.

File: components/director/pkg/persistence/persistence.go [hunk 2]
```diff
@@ -45,9 +46,14 @@ type db struct {
 }
 
 func (db *db) PingContext(ctx context.Context) error {
+	db.sqlDB.Stats()
```
Comment (crabtree) [inline components/director/pkg/persistence/persistence.go:49]: What is the purpose of this line here?

File: components/director/cmd/director/main.go (hunk 1 above)
Comment (pkosiec) [reply components/director/cmd/director/main.go:314]: Very good idea, I will do it

File: components/director/pkg/persistence/persistence.go (hunk 2 above)
Comment (pkosiec) [reply components/director/pkg/persistence/persistence.go:49]: Sorry, this shouldn't be here. I will fix it in a moment.

File: components/director/README.md [hunk 3]
```diff
@@ -68,11 +69,11 @@ The Director binary allows to override some configuration parameters. You can sp
 | **APP_CONNECTOR_URL**                        | None                            | The endpoint of Connector                                          |
 | **APP_OAUTH20_CLIENT_ENDPOINT**              | None                            | The endpoint for managing OAuth 2.0 clients                        |
 | **APP_OAUTH20_PUBLIC_ACCESS_TOKEN_ENDPOINT** | None                            | The public endpoint for fetching OAuth 2.0 access token            |
+| **APP_OAUTH20_HTTP_CLIENT_TIMEOUT**          | `3m`                            | The timeout of HTTP Client for managing OAuth 2.0 clients          |
```
Comment (klaudiagrz) [inline components/director/README.md:72]:
```suggestion
| **APP_OAUTH20_HTTP_CLIENT_TIMEOUT**          | `3m`                            | The timeout of HTTP client for managing OAuth 2.0 clients          |
```

**Pull request merged**